app/core/llm_framework/openai_langchain.py

```diff
@@ -25,7 +25,7 @@ class LangchainOpenAI(LLMFrameworkInterface):
     chain = None
     vectordb = None
     def __init__(self, #pylint: disable=super-init-not-called
-                key:str=os.getenv("OPENAI_API_KEY"),
```

Comment:
I wonder if it might make sense to just have a `env.test` file that we load?

Reply:
This is the only one I would think about, but it's not a big blocker if we don't want to do it now. Maybe just add a FIXME or something if not.

Reply:
This is something I struggled with a lot...
What I actually tried to do was to mock the constructors of OpenAI and Supabase classes where they connect to the API service. This way, even if we have an API key saved in our env, we wont be charged accidentaly while running tests. But  for some reason couldn't get it to work. (I am using mock for the first time. )

I want to come back to it if I can... Will add a fixme.